Sources/llbuild2/FunctionCache/FileBackedFunctionCache.swift

```diff
@@ -22,23 +22,28 @@ public final class LLBFileBackedFunctionCache: LLBFunctionCache {
     let threadPool: NIOThreadPool
     let fileIO: NonBlockingFileIO
 
+    /// Salt value to mix into the path of the file. To be used by clients for "namespacing" the caching of keys, for
+    /// example, by setting this to a version number or the hash of the binary. This way, if the client decides to
+    /// change this salt, the caching namespace will be entirely different.
+    private let salt: UInt64
 
     /// Create an in-memory database.
-    public init(group: LLBFuturesDispatchGroup, path: AbsolutePath) {
+    public init(group: LLBFuturesDispatchGroup, path: AbsolutePath, salt: UInt64 = 0) {
         self.group = group
         self.threadPool = NIOThreadPool(numberOfThreads: 6)
         threadPool.start()
         self.fileIO = NonBlockingFileIO(threadPool: threadPool)
         self.path = path
         try? localFileSystem.createDirectory(path, recursive: true)
+        self.salt = salt
     }
 
     deinit {
         try? threadPool.syncShutdownGracefully()
     }
 
     private func filePath(key: LLBKey) -> AbsolutePath {
-        return path.appending(component: "\(key.stableHashValue)")
+        return path.appending(component: "\(salt)_\(key.stableHashValue)")
```

Comment:
probably nicer to use salt as a subpath `path.appending(components: salt, key.stableHashValue)`

Reply:
renamed to version and did exactly that in the initializer